shared/wallets/confirm-form/container.js

```diff
@@ -40,13 +40,29 @@ const mapStateToProps = state => {
 }
 
 const mapDispatchToProps = (dispatch, {navigateUp, onBack}: OwnProps) => ({
-  onBack: () => (navigateUp ? dispatch(navigateUp()) : onBack && onBack()),
-  onClose: () => (navigateUp ? dispatch(navigateUp()) : onBack && onBack()),
+  _onBack: () => (navigateUp ? dispatch(navigateUp()) : onBack && onBack()),
+  _onClose: () => (navigateUp ? dispatch(navigateUp()) : onBack && onBack()),
+  _onClearErrors: () => dispatch(WalletsGen.createClearErrors()),
   onSendClick: () => dispatch(WalletsGen.createSendPayment()),
 })
 
 export default connect(
   mapStateToProps,
   mapDispatchToProps,
-  (s, d, o) => ({...o, ...s, ...d})
+  (s, d, o) => {
+    // Clear sentPaymentError when navigating away.
```

Comment:
we should move this logic into the sagas and not these individual components

Reply:
we have a couple of examples of this, look in devices or provision or signup for navigateup handling

Reply:
So signup has the closest thing to what I want, which is `goBackAndClearErrors`. However, one complication is that signup always can do a `navigateUp`, whereas here we do `navigateUp` on mobile but we have to call `onBack` which changes the state of the `SendForm` component.

I can't quite see how to put all the logic in saga when `onBack` has to be something that touches a component's state. What should I do?

Reply:
Ok let's leave this for now. When we revisit nav I think this needs to be a first class concept (aka exiting a route)

Reply:
Actually maybe it's better to move this to the root and have onback always passes in. Then the root of the form which decides on the routing (mobile or not) passes on back which changes local state or does a navigate up. That then also clears errors. This way all sub views get this behavior for free and they don't need to know about the routing

Reply:
this could be done with overwriting navigateUp etc in routeprops but likely not worth doing, we can keep this as-is but its a pattern we need a simpler solution for in the future

Reply:
i talked through a solution to this w/ danny. i'm making a ticket to remove the divide between mobile and desktop and instead it'll all be route based